### DISCUSSION TITLE
WebUI > Information > MQTT Group Topic 1: shows tasmotas/[...]

### DISCUSSION THREAD

**Discussion by cmonty14** [arendst/Tasmota#22191] [Support request]
Hi,
I have flashed latest Tasmota 14.2.0.4 (d5289e8-tasmota) on my power plugs [Gosund SP111 v1.1](https://templates.blakadder.com/gosund_SP111_v1_1.html).
Everything runs normally.

However, in _WebUI > Information > MQTT Group Topic 1_ I can see this:
`tasmotas/plug/gosund-sp111/4/cmnd/`
![Bildschirmfoto vom 2024-09-19 20-17-52](https://github.com/user-attachments/assets/66f2bb04-7f77-4db0-9faf-f97ee838d755)

However, on another power plug [Blitzwolf SHP2](https://templates.blakadder.com/blitzwolf_SHP2.html) the same item is showing this value:
`tasmota/plug/blitzwolf-shp2/1/cmnd/`

Could you please advise how to fix this?

THX

**Comment by sfromis** [comment]
There is nothing to be fixed there, except for your not very good way of using the fields.

Tasmota has a default group topic of `tasmotas`, which is then merged with your setting for full topic, resulting in what you see there. Looks like you have used the full topic setting in a way it is not meant to, as the full topic is **not** meant to be individual per  device. And your setting for "Topic" also not very good, as this should identify each device, and then the `%topic%` part of the full topic you set becomes what you have set in topic, or in grouptopic.

What fulltopic is meant for is to change the structure of the MQTT topics, and your full topic should be the same for many (or all) devices, with the topic setting providing the variation. And then you can use the group topic to send commands to all devices having the same group topic.

**Comment by cmonty14** [comment]
I'm sorry, but I don't get it...

Could you please advise what are recommended values for:

- Client
- Topic
- Full Topic

I have configured `tasmota/#` as _MQTT Subscriptions_ in my _MQTT Gateway_.
![Bildschirmfoto vom 2024-09-24 17-32-29](https://github.com/user-attachments/assets/c686d170-108e-4ad7-813f-caa158699dad)

**Reply by sfromis** [reply]
As a starting point, don't mess with MQTT client, unless you know why you want it. In most cases, not very useful.

What you show in that screenshot looks ok, with a device-specific "Topic" and the default "Full Topic". 
You could use the latter to enact a common topic structure, like `mydevices/%prefix%/%topic%/`, or other variations, as long as you have both `%prefix%` and `%topic%` included.

The reasonable MQTT subscription to match would be `#`, or with my example `+/mydevices/#`.
Your screenshot should give topics like `tele/tasmota_blitzwolf-shp2-1/STATE`, and you can publish commands to `cmnd/tasmota_blitzwolf-shp2-1/<commandname>`.

You can have up to 4 group topics, where a `GroupTopic` of `mygroup` allows commands to multiple devices `cmnd/mygroup/<commandname>`. Likewise with the default grouptopic of `tasmotas`. Group topics are not relevant to subscribe to, only to publish commands to, for all devices subscribing to the same group topics.

If you set `Weblog 3` and restart, some of the first console messages will show what topics Tasmota is subscribing to, including the effects of your group topics.

**Comment by cmonty14** [comment]
Actually the subscription required on my MQTT Gateway is: `stat/#`

This is somehow unlucky as I would prefer subscription `tasmotas/#`

**Reply by sfromis** [reply]
The reason why I suggested `#` was that you do not get telemetry data with `stat/#`. Of course, you could subscribe to both `tele/#` and `stat/#` - or amend the full topic to include a common qualifier, which indeed *could* be `tasmotas` (instead of my example of `mydevices`). Just make sure to not confuse your topic starter with the default group topic of `tasmotas` (or my example of `mygroup`).

**Reply by barbudor** [reply]
> Actually the subscription required on my MQTT Gateway is: `stat/#`
> 
> This is somehow unlucky as I would prefer subscription `tasmotas/#`

I'm not sure what you mean by MQTT gateway.

But group topics are meant only to send commands (cmnd) to a group of devices.
Responses (stat) or telemetry (tele) are always sent by devices to their individual topic.

When sending a STATUS command to the group topic, it would be messy if each device was sending back it's status on the same topic
